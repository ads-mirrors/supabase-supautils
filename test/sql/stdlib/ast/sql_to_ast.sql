select supa.sql_to_ast($$
    create table acct(
        id int primary key
    )
$$::text);
