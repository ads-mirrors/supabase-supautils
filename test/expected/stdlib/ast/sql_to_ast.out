select supa.sql_to_ast($$
    create table acct(
        id int primary key
    )
$$::text);
                sql_to_ast                
------------------------------------------
 (                                       +
    {RAWSTMT                             +
    :stmt                                +
       {CREATESTMT                       +
       :relation                         +
          {RANGEVAR                      +
          :schemaname <>                 +
          :relname acct                  +
          :inh true                      +
          :relpersistence p              +
          :alias <>                      +
          :location 18                   +
          }                              +
       :tableElts (                      +
          {COLUMNDEF                     +
          :colname id                    +
          :typeName                      +
             {TYPENAME                   +
             :names ("pg_catalog" "int4")+
             :typeOid 0                  +
             :setof false                +
             :pct_type false             +
             :typmods <>                 +
             :typemod -1                 +
             :arrayBounds <>             +
             :location 35                +
             }                           +
          :inhcount 0                    +
          :is_local true                 +
          :is_not_null false             +
          :is_from_type false            +
          :storage <>                    +
          :raw_default <>                +
          :cooked_default <>             +
          :identity <>                   +
          :identitySequence <>           +
          :generated <>                  +
          :collClause <>                 +
          :collOid 0                     +
          :constraints (                 +
             {CONSTRAINT                 +
             :conname <>                 +
             :deferrable false           +
             :initdeferred false         +
             :location 39                +
             :contype PRIMARY_KEY        +
             :keys <>                    +
             :including <>               +
             :options <>                 +
             :indexname <>               +
             :indexspace <>              +
             :reset_default_tblspc false +
             }                           +
          )                              +
          :fdwoptions <>                 +
          :location 32                   +
          }                              +
       )                                 +
       :inhRelations <>                  +
       :partspec <>                      +
       :partbound <>                     +
       :ofTypename <>                    +
       :constraints <>                   +
       :options <>                       +
       :oncommit 0                       +
       :tablespacename <>                +
       :accessMethod <>                  +
       :if_not_exists false              +
       }                                 +
    :stmt_location 0                     +
    :stmt_len 0                          +
    }                                    +
 )                                       +
 
(1 row)

